### DISCUSSION TITLE
PDF with Font Linking

### DISCUSSION THREAD

**Discussion by miguelelvir** [QuestPDF/QuestPDF#748] [Q&A] [answered]
Hi @all!
We are working on supporting reports with international characters in it. Does anyone have experience using QuestPDF to generate a file that uses [font linking](https://learn.microsoft.com/en-us/globalization/fonts-layout/fonts#font-linking-and-gdi) instead of font embedding? I noticed we already have an [issue thread](https://github.com/QuestPDF/QuestPDF/issues/31) relating to a similar issue when embedding large fonts. However, I wonder if we also support font linking as an option?

**Comment by MarcinZiabek** [comment]
The issue with large fonts will be finally resolved by automated font subsetting introduced as part of this effort https://github.com/QuestPDF/QuestPDF/discussions/622 The release is rather sooner than later :)

**Reply by miguelelvir** [reply]
Thanks @MarcinZiabek for the wonderful job you are doing with the library. We are looking forward to font subsetting as an option, since this could also cover our needs. Just to put the nail in our investigative coffin if needed, would you say there will not be direct support for font linking moving forward and that the recommended way will be to use font subsetting?

**Reply by MarcinZiabek** [reply, marked as answer]
I confirm that font subsetting is the current path moving forward. I don't plan on supporting font linking at this moment - it is technically possible by extending internal Skia code, but also tricky to maintain.

I can confirm that font subsetting and document content compression can do wonders when it comes to PDF file size 😀

**Comment by MarcinZiabek** [comment]
Hello,

We are closing this thread due to inactivity.

To confirm the direction noted above: font linking is not planned, and automated font subsetting is the supported path — that work has since been released, so embedded international fonts are reduced to only the glyphs actually used. Combined with [document compression](https://www.questpdf.com/concepts/document-settings.html) (CompressDocument), this keeps output size small without needing font linking.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
